diff -Nrup a/include/libtorrent/span.hpp b/include/libtorrent/span.hpp
--- a/include/libtorrent/span.hpp	2019-09-24 03:52:01.000000000 -0600
+++ b/include/libtorrent/span.hpp	2019-10-29 11:21:24.793750340 -0600
@@ -33,6 +33,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_SPAN_HPP_INCLUDED
 #define TORRENT_SPAN_HPP_INCLUDED
 
+#include <string>
 #include <array>
 #include <type_traits>
 #include "libtorrent/assert.hpp"
