--- libtorrent-rasterbar-0.14.10/examples/Makefile.am.orig	2010-05-28 08:37:02.290439097 +0530
+++ libtorrent-rasterbar-0.14.10/examples/Makefile.am	2010-05-28 08:37:21.397439025 +0530
@@ -6,8 +6,7 @@
 EXTRA_DIST = Jamfile
 
 client_test_SOURCES = client_test.cpp
-client_test_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la @BOOST_REGEX_LIB@ @BOOST_PROGRAM_OPTIONS_LIB@
-
+client_test_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la @BOOST_REGEX_LIB@ @BOOST_PROGRAM_OPTIONS_LIB@ -lrt
 dump_torrent_SOURCES = dump_torrent.cpp
 dump_torrent_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la
 
