--- libtorrent-rasterbar-1.1.13/include/libtorrent/config.hpp.orig	2019-04-27 10:25:22.000000000 -0500
+++ libtorrent-rasterbar-1.1.13/include/libtorrent/config.hpp	2019-05-03 08:56:01.793014380 -0500
@@ -50,6 +50,7 @@
 #include <boost/config.hpp>
 #include <boost/asio/detail/config.hpp>
 #include <boost/version.hpp>
+#include <boost/core/noncopyable.hpp>
 #include <stdio.h> // for snprintf
 #include <limits.h> // for IOV_MAX
 
