--- libtorrent-rasterbar-0.14.10/configure.py27	2010-03-21 07:03:10.000000000 +0900
+++ libtorrent-rasterbar-0.14.10/configure	2010-07-29 03:23:36.000000000 +0900
@@ -18486,7 +18486,7 @@
 $as_echo_n "checking for python build information... " >&6; }
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: " >&5
 $as_echo "" >&6; }
-for python in python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
+for python in python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
 # AC_CHECK_PROGS(PYTHON_BIN, [$python])
 PYTHON_BIN=$python
 ax_python_bin=$PYTHON_BIN
