diff -up ./src/file.cpp.old ./src/file.cpp
--- ./src/file.cpp.old	2008-02-14 20:53:03.000000000 -0800
+++ ./src/file.cpp	2008-02-14 20:53:14.000000000 -0800
@@ -66,6 +66,7 @@ BOOST_STATIC_ASSERT(sizeof(lseek(0, 0, 0
 #include <boost/filesystem/operations.hpp>
 #include "libtorrent/file.hpp"
 #include <sstream>
+#include <cstring>
 
 #ifndef O_BINARY
 #define O_BINARY 0
diff -up ./include/libtorrent/buffer.hpp.old ./include/libtorrent/buffer.hpp
--- ./include/libtorrent/buffer.hpp.old	2008-02-14 20:50:39.000000000 -0800
+++ ./include/libtorrent/buffer.hpp	2008-02-14 20:50:59.000000000 -0800
@@ -36,6 +36,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/invariant_check.hpp"
 #include <memory>
+#include <cstring>
 
 namespace libtorrent {
 
diff -up ./include/libtorrent/peer_id.hpp.old ./include/libtorrent/peer_id.hpp
--- ./include/libtorrent/peer_id.hpp.old	2008-02-14 20:53:41.000000000 -0800
+++ ./include/libtorrent/peer_id.hpp	2008-02-14 20:54:02.000000000 -0800
@@ -39,6 +39,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <cctype>
 #include <algorithm>
 #include <string>
+#include <cstring>
 
 #include "libtorrent/config.hpp"
 
diff -up ./include/libtorrent/file.hpp.old ./include/libtorrent/file.hpp
diff -up ./include/libtorrent/asio/detail/pipe_select_interrupter.hpp.old ./include/libtorrent/asio/detail/pipe_select_interrupter.hpp
--- ./include/libtorrent/asio/detail/pipe_select_interrupter.hpp.old	2008-02-14 20:49:42.000000000 -0800
+++ ./include/libtorrent/asio/detail/pipe_select_interrupter.hpp	2008-02-14 20:50:09.000000000 -0800
@@ -32,6 +32,8 @@
 #include "asio/system_error.hpp"
 #include "asio/detail/socket_types.hpp"
 
+#include <cstring>
+
 namespace asio {
 namespace detail {
 
