--- libtorrent-rasterbar-0.14.10/examples/Makefile.in.orig	2010-05-28 08:38:35.060439029 +0530
+++ libtorrent-rasterbar-0.14.10/examples/Makefile.in	2010-05-28 08:40:31.017689483 +0530
@@ -263,7 +263,7 @@
 top_srcdir = @top_srcdir@
 EXTRA_DIST = Jamfile
 client_test_SOURCES = client_test.cpp
-client_test_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la @BOOST_REGEX_LIB@ @BOOST_PROGRAM_OPTIONS_LIB@
+client_test_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la @BOOST_REGEX_LIB@ @BOOST_PROGRAM_OPTIONS_LIB@ -lrt
 dump_torrent_SOURCES = dump_torrent.cpp
 dump_torrent_LDADD = $(top_builddir)/src/libtorrent-rasterbar.la
 make_torrent_SOURCES = make_torrent.cpp
